### ISSUE TITLE
Reading variants takes longer than a minute, possible to only read subset of variants?

### ISSUE THREAD

**Issue by ParsaAkbari** [limix/bgen-reader-py#6]
I want to read a subset of variants from a large BGEN file because reading the entire BGEN file using read_bgen() is very time consuming.

Is it possible to adapt the code in a way which would make this possible?

Thanks for your help.

**Comment by horta** [comment]
It is supposed to load only the variants you request. How are you using it to select the variants?

**Comment by ParsaAkbari** [comment]
Thanks for the response.

```
# this step seems to be taking a while which is motivating me to subset the BGEN files before loading into bgen-reader
bgen_file = read_bgen(bgen_path, verbose=False)

# read_variants contains all the variants in the bgen file, I then have to subset this down
read_variants = bgen_file['variants']

I can then extract the genotype of individual variants using the
bgen_file['genotype']....compute() method

```
Apologies if I am missing something here, thanks for your help!

**Comment by horta** [comment]
I had implemented a cache in bgen 1.1.0 (https://github.com/limix/bgen/releases) that makes subsequent loading way faster but I haven't finished the conda build for it yet (https://github.com/conda-forge/bgen-feedstock/pull/12) nor finished a Python wrapper for that. I will finish it this weekend. Please, remind me if I don't get back to you before Sunday!
